pkg/storage/storage.go

```diff
@@ -293,3 +294,35 @@ func ChunkType(ch swarm.Chunk) swarm.ChunkType {
 	}
 	return swarm.ChunkTypeUnspecified
 }
+
+// IdentityAddress returns the internally used address for the chunk
+// since the single owner chunk address is not a unique identifier for the chunk,
+// but hashing the soc address and the wrapped chunk address is.
+// it is used in the reserve sampling and other places where a key is needed to represent a chunk.
+func IdentityAddress(chunk swarm.Chunk) (swarm.Address, error) {
```

Comment:
this is tricky to do because IdentityAddress calculation calls exported functions of the cac and soc packages, in return the cac and soc packages use the swarm package. Because of circular dependency, it's currently not possible to do your suggestion without some refactoring.